### PR TITLE
Expose methods to operate on vectors through bindings 

### PR DESCRIPTION
**Pull request by aymanhab**
Fixes issue #0

### CHANGELOG.md (choose one)

- updated...

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2558)

### CODE REVIEW

**Comment by aymanhab** [comment]
@chrisdembia @tkuchida and @jimmyDunne This was inspired by seeing code that jumps through hoops to divide a Vec3 by a constant in Matlab. Don't know if we should enable the methods for all Vec<x> types and also what other methods would be helpful. Feedback is appreciated. This should reduce the over head significantly.

**Comment by aymanhab** [comment]
This should also work for python @chrisdembia @nickbianco unless you found a better way to expose operators already.

**Comment by tkuchida** [comment]
> This should also work for python @chrisdembia @nickbianco unless you found a better way to expose operators already.

In Python, you can do the following (see [Stack Overflow](https://stackoverflow.com/questions/50599045/python-replacing-a-function-within-a-class-of-a-module)):

```
import opensim as osim

def myVec3Add(self,v):
    newvec = osim.Vec3(self[0]+v[0], self[1]+v[1], self[2]+v[2])
    return newvec

osim.Vec3.__add__ = myVec3Add

a = osim.Vec3(1,2,3)
b = osim.Vec3(4,5,6)
c = a+b
```

**Comment by chrisdembia** [comment]
Nice!

File: Bindings/Java/tests/TestVectors.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import org.opensim.modeling.*;
+
+class TestVectors {
+    public static void TestVectors() {
+        Vec3 m = new Vec3(1, 2, 3);
+        System.out.println(m.toString());
+        m.scalarEq(5.);
+        System.out.println(m.toString());
+	m.scalarPlusEq(3.);
```
Comment (chrisdembia) [inline Bindings/Java/tests/TestVectors.java]: Fix spacing?

File: Bindings/Java/tests/TestVectors.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	Vec3 n = new Vec3(8);
+        System.out.println(n.toString());
+	n.scalarTimesEq(1000.0);
+	n.scalarDivideEq(1000.0);
+	// Check m & n are same = {8, 8, 8}
+	for (int i=0; i<3; i++)
+	   assert(m.get(i) == n.get(i));
```
Comment (chrisdembia) [inline Bindings/Java/tests/TestVectors.java]: This test would pass if both functions do nothing.

Comment (aymanhab) [reply Bindings/Java/tests/TestVectors.java]: This is just place holder to make sure the methods are available. Will turn into real test based on answers to questions above. Thanks for the prompt review @chrisdembia

File: Bindings/Java/tests/TestVectors.java (hunk 1 above)
Comment (aymanhab) [reply Bindings/Java/tests/TestVectors.java]: Done

**Comment by aymanhab** [comment]
@chrisdembia I addressed your feedback, thanks for the prompt review. More feedback on the issues in my first comment would be much appreciated as I think this is a safe change that can have significant positive impact to include if merged. Thank you

**Comment by aymanhab** [comment]
Thanks @chrisdembia

**Pull request merged**